Sources/AsyncQueue/ActorQueue.swift

```diff
@@ -0,0 +1,144 @@
+        // If this force-unwrap fails, something is fundamentally broken in the Swift runtime.
+        taskStreamContinuation = capturedTaskStreamContinuation!
+
+        Task.detached(priority: priority) {
+            let executor = ActorExecutor()
+            for await task in taskStream {
+                await executor.suspendUntilStarted(task)
+            }
+        }
+    }
+
+    deinit {
+        taskStreamContinuation.finish()
+    }
+
+    // MARK: Public
+
+    /// Schedules an asynchronous task for execution and immediately returns.
+    /// The scheduled task will not execute until all prior tasks have completed or suspended.
+    /// - Parameter task: The task to enqueue.
+    public func async(_ task: @escaping @Sendable () async -> Void) {
+        taskStreamContinuation.yield(task)
+    }
+
+    /// Schedules an asynchronous task and returns after the task is complete.
+    /// The scheduled task will not execute until all prior tasks have completed or suspended.
+    /// - Parameter task: The task to enqueue.
+    /// - Returns: The value returned from the enqueued task.
+    public func await<T>(_ task: @escaping @Sendable () async -> T) async -> T {
+        await withUnsafeContinuation { continuation in
+            taskStreamContinuation.yield {
+                continuation.resume(returning: await task())
+            }
+        }
+    }
+
+    /// Schedules an asynchronous throwing task and returns after the task is complete.
+    /// The scheduled task will not execute until all prior tasks have completed or suspended.
+    /// - Parameter task: The task to enqueue.
+    /// - Returns: The value returned from the enqueued task.
+    public func await<T>(_ task: @escaping @Sendable () async throws -> T) async throws -> T {
+        try await withUnsafeThrowingContinuation { continuation in
+            taskStreamContinuation.yield {
+                do {
+                    continuation.resume(returning: try await task())
+                } catch {
+                    continuation.resume(throwing: error)
+                }
+            }
+        }
+    }
+
+    // MARK: Private
+
+    private let taskStreamContinuation: AsyncStream<@Sendable () async -> Void>.Continuation
+
+    // MARK: - ActorExecutor
+
+    private actor ActorExecutor {
+        func suspendUntilStarted(_ task: @escaping @Sendable () async -> Void) async {
+            // Suspend the calling code until our enqueued task starts.
+            await withUnsafeContinuation { continuation in
+                // Utilize the serial (but not FIFO) Actor context to execute the task without requiring the calling method to wait for the task to complete.
+                Task {
+                    // Force this task to execute within the ActorExecutor's context by accessing an ivar on the instance.
+                    // Without this line the task executes on a random context, causing execution order to be nondeterministic.
+                    _ = void
```

Comment:
I've been able to reproduce this issue on tvOS and macOS, but not iOS. I haven't tested watchOS just yet. Looking into whether this is a known issue. Otherwise I'll file a bug.